transport_nantes/topicblog/templates/topicblog/base_topicblog.html

```diff
@@ -10,42 +10,98 @@
 Maybe there's a better way, I just haven't found it yet.
 {% endcomment %}
 
-{% block body %}{% if user.is_staff %}
-  <div class="container-fluid pl-3 py-1">{% visit_overview %}</div>
-{% endif %}
-{% if user.is_staff and topicblog_admin  %}
-<div class="container-fluid pl-5 py-3">
-    <div class="row">
-	<p>
-	    <a type="button" class="btn btn-outline-info btn-sm"
-	       href="{% url 'topic_blog:list_items_by_slug' page.slug|default:'--sans-slug--' %}">Liste</a>
-	    {% if page.publication_date %}
-	    	publiée {{ page.publication_date|date:"l d-m-Y" }} par {{ page.publisher.email }}
-	    {% else %}
-			{% if page.first_publication_date %}
-				publiée {{ page.first_publication_date|date:"l d-m-Y" }} par {{ page.publisher.email }} (retirée)
+{% block body %}
+{% block common_staff_banner %}
+	{% if user.is_staff %}
+		<div class="container-fluid pl-3 py-1">{% visit_overview %}</div>
+	{% endif %}
+	{% if user.is_staff and topicblog_admin  %}
+	<div class="container-fluid pl-5 py-3">
+		<div class="row">
+		<p>
+			<a type="button" class="btn btn-outline-info btn-sm"
+			href="{% url page.listone_object_url page.slug|default:'--sans-slug--' %}">Liste</a>
+			{% if page.publication_date %}
+				publié {{ page.publication_date|date:"l d-m-Y" }} par {{ page.publisher.email }}
 			{% else %}
-				(brouillon)
+				{% if page.first_publication_date %}
+					publié {{ page.first_publication_date|date:"l d-m-Y" }} par {{ page.publisher.email }} (retiré)
+				{% else %}
+					(brouillon)
+				{% endif %}
+				{% if page.is_publishable %}
+					<form
+					action="{% url page.viewbypkid_object_url pkid page.slug|default:'--sans-slug--' %}"
+					method="POST"
+					class="pl-1">
+						{% csrf_token %}
+						<input type="submit" class="btn btn-outline-info btn-sm" value="Publier">
+					</form>
+				{% endif %}
 			{% endif %}
-			{% if page.is_publishable %}
-				<form
-				action="{% url 'topicblog:view_item_by_pkid' pkid page.slug|default:'--sans-slug--' %}"
-				method="POST">
-					{% csrf_token %}
-					<input type="submit" class="btn btn-outline-info btn-sm" value="Publier">
-				</form>
-			{% endif %}
-	    {% endif %}
 
-	    <a type="button" class="btn btn-outline-info btn-sm"
-	       href="{{ page.get_edit_url }}">Modifier</a>
+			<div class="px-1">
+				<a type="button" class="btn btn-outline-info btn-sm"
+			href="{{ page.get_edit_url }}">Modifier</a>
+			</div>
+
+			<div class="px-1">
+				<a type="button" class="btn btn-outline-info btn-sm"
+				href="{% url page.viewbyslug_object_url page.slug|default:'--sans-slug--' %}">
+					Visualiser (usager)
+				</a>
+			</div>
 
-	    {% if page.publication_date %}
-	    <a type="button" class="btn btn-outline-info btn-sm"
-	       href="{% url 'topic_blog:view_item_by_slug' page.slug|default:'--sans-slug--' %}">Visualiser (usager)</a>
-	    {% endif %}
-	</p>
-    </div>
-</div>
-{% endif %}{{ block.super }}
+			{% if transactional_send_record_class %}
```

Comment:
I think this should be done with a second template that inherits from this one and simply modifies this bit, possibly by calling `block.super` and then adding something.  Imagine that we have another thing that's sort of like sendable but different, and then we add that block here.  And then another.  This template shouldn't know about all the ways it might be used.

Reply:
but it's a good idea to generalise it!